src/server/document-controller.js

```diff
@@ -129,6 +130,26 @@ class DocumentController {
       return res.send({ documentRevisions });
     });
 
+    router.post('/api/v1/docs/:key/archive', needsPermission(permissions.MANAGE_ARCHIVED_DOCS), async (req, res) => {
```

Comment:
Why not go with the normal way of updating the documents? Making 2 new endpoints for this seems excessive :)

Reply:
Cause I don't want to pass in all the data of the new revision, just the value of the flag 🤷 Less error prone also.

Reply:
Feels like a PATCH operation to me, but okay.

Reply:
Ok so you meant from verb perspective not path. I can change to PATCH, true

Reply:
gePATCHed. I kept 2 endpoints though, as we have multiple routes starting with `/api/v1/docs/` and wanted to make sure there'll be no clashed now (if the order of the declarations is not correct) or in the future

Reply:
Sounds good to me. The rest of the code is not very religious about REST though, especially as we do not really patch anyway (it's always a new revision, not an in-place update).

Reply:
Yes true, basically we always POST